.github/workflows/composition.yml

```diff
@@ -20,10 +23,10 @@ jobs:
           restore-key: ${{ runner.os }}-${{ runner.arch }}-cargo-dev
 
       - run: cargo clippy
-        working-directory: 'crates/composition'
+        working-directory: crates/${{ matrix.package }}
 
       - run: cargo test
-        working-directory: 'crates/composition'
+        working-directory: crates/${{ matrix.package }}
       
       - run: cargo fmt --check
-        working-directory: 'crates/composition'
+        working-directory: crates/${{ matrix.package }}
```

Comment:
Now thinking maybe we should have some duplication and run these sequentially, it seems wasteful to spin up two runners for that.